### ISSUE TITLE
clipboard not working on docker selfhosted app

### ISSUE THREAD

**Issue by clacalderonc** [cogentapps/chat-with-gpt#170]
Hi Team,

When i use the self hosted version of the app, the clipboard button for chat content (or markdown code blocks) doesn't really work at all.

Self hosted:
![image](https://github.com/cogentapps/chat-with-gpt/assets/46971883/6afb99a4-a317-45f4-934d-a56cf0b4c84e)

chatwithgpt.ai

![image](https://github.com/cogentapps/chat-with-gpt/assets/46971883/60e8c16f-7be7-458d-a2e7-01230ffa976c)

i'm using the latest docker image available, tried on chrome, edge, firefox, same result, no errors con browser console.

Rergards.
Claudio

**Comment by cogentapps** [comment]
Thanks for reporting this. Just to confirm, are you using the release branch of the docker image (docker pull ghcr.io/cogentapps/chat-with-gpt:release)?

**Comment by clacalderonc** [comment]
> Thanks for reporting this. Just to confirm, are you using the release branch of the docker image (docker pull ghcr.io/cogentapps/chat-with-gpt:release)?

yes, using the release tag and just to be sure I recreated the environment several times, i made this small docker-compose to handle my env.

```
version: '3'
services:
  app:
    image: ghcr.io/cogentapps/chat-with-gpt:release
    ports:
      - 3000:3000
    volumes:
      - ./data:/app/data

**Comment by cogentapps** [comment]
Most likely because many browsers block copying on 'insecure' connections (no SSL) and the Docker command doesn't use SSL by default. We have experimental support for enabling this in ./data/config.yaml:

```
tls:
  selfSigned: true
```

but it’s currently untested and may not work. See discussion in #132 for another possible approach.

**Comment by clacalderonc** [comment]
> Most likely because many browsers block copying on 'insecure' connections (no SSL) and the Docker command doesn't use SSL by default. We have experimental support for enabling this in ./data/config.yaml:
> 
> ```
> tls:
>   selfSigned: true
> ```
> 
> but it’s currently untested and may not work. See discussion in #132 for another possible approach.

Great, I enabled that option and issued a letsencrypt certificate and is working now, I saw  that the selfSigned option try to use a script that is not present on the image ([generate-self-signed-certificate.sh](https://github.com/cogentapps/chat-with-gpt/blob/main/server/generate-self-signed-certificate.sh)).

Anyway, you can put cert.pem / key.pem files in  data/ directory, plus the self signed option enabled and ssl it's working.
